scipy/_lib/_array_api_docs_tables.py

```diff
@@ -0,0 +1,226 @@
+        "js_roots",
+    }
+}
+
+
+def _process_capabilities_table_entry(entry: dict | None) -> dict[str, str]:
+    """Returns flat dict showing what is and isn't supported in entry."""
+    if entry is None:
+        # If there is no entry, assume no alternative backends are supported.
+        # If the list of supported backends will grows, this hard-coded dict
+        # will need to be updated.
+        return {
+            "cupy": False, "torch cpu": False,
+            "torch gpu": False, "jax.numpy cpu": False,
+            "jax.numpy gpu": False, "jax.numpy jit": False,
+            "dask.array": False, "dask.array lazy": False
+        }
+    row = {}
+    # For now, use _make_sphinx_capabilities because that's where
+    # the relevant logic for determining what is and isn't
+    # supported based on xp_capabilities_table entries lives.
+    # Perhaps this logic should be decoupled from sphinx.
+    for backend, capabilities in _make_sphinx_capabilities(**entry).items():
+        if backend in {"array_api_strict", "numpy"}:
+            continue
+        cpu, gpu = capabilities.cpu, capabilities.gpu
+        if cpu is None:
+            row[backend] = capabilities.gpu
+        elif gpu is None:
+            row[backend] = capabilities.cpu
+        else:
+            row[f"{backend} cpu"] = capabilities.cpu
+            row[f"{backend} gpu"] = capabilities.gpu
+        if backend == "jax.numpy":
+            row["jax.numpy jit"] = entry["jax_jit"] and row["jax.numpy cpu"]
+        if backend == "dask.array":
+            support_lazy = not entry["allow_dask_compute"] and row["dask.array"]
+            row["dask.array lazy"] = support_lazy
+    return row
+
+
+def make_flat_capabilities_table(
+        modules: str | list[str],
+        /,
+        *,
+        capabilities_table: list[str] | None = None,
+) -> list[dict[str, str]]:
+    """Generate full table of array api capabilities across public functions.
+
+    Parameters
+    ----------
+    modules : str | list[str]
+        A string containing single SciPy module, (e.g `scipy.stats`, `scipy.fft`)
+        or a list of such strings.
+
+    Returns
+    -------
+    output : list[dict[str, str]]
+        `output` is a table in dict format
+        (keys corresponding to column names). If `modules` is a list, then the
+        first column will be "module" If `modules` is a string, then the
+        "module" column will be omitted. The other colums are currently
+    
+        "cupy", "torch (cpu)", "torch (gpu)", "jax.numpy (cpu)", "jax.numpy (gpu)",
+        "jax.numpy (jit)", "dask.array", "dask.array (lazy)".
+
+        "numpy" is omitted because it will always be supported.
+    """
+    multiple_modules = True
+    if isinstance(modules, str):
+        modules = [modules]
+        multiple_modules = False
+    if capabilities_table is None:
+        capabilities_table = xp_capabilities_table
+
+    output = []
+
+    from scipy.stats._distn_infrastructure import rv_generic
```

Comment:
You mentioned this in your last message, but I think there needs to be another way. (Maybe even looking at the `__name__` attribute of `__thing__`'s class, if there's nothing else?)

Reply:
Yeah, importing from `stats` to do this here is atrocious. I've gone back to skipping unnamed callables.